chartfx-chart/src/main/java/io/fair_acc/chartfx/axes/spi/format/DefaultFormatter.java

```diff
@@ -77,84 +66,142 @@ public DefaultFormatter(final Axis axis, final String prefix, final String suffi
     public Number fromString(final String string) {
         final int prefixLength = prefix == null ? 0 : prefix.length();
         final int suffixLength = suffix == null ? 0 : suffix.length();
+        final String valueString = string.substring(prefixLength, string.length() - suffixLength);
         try {
-            return formatterSmall.parse(string.substring(prefixLength, string.length() - suffixLength));
-        } catch (final ParseException exc) {
-            try {
-                return formatterLarge.parse(string.substring(prefixLength, string.length() - suffixLength));
-            } catch (final ParseException ex) {
-                ex.addSuppressed(exc);
-                throw new IllegalArgumentException(ex);
-            }
+            // TODO: The previous implementation used the DecimalFormat for parsing. Did that do anything special?
+            return formatter.fromString(valueString);
+        } catch (final Throwable exc) {
+            // Matches the previous contract
+            throw new IllegalArgumentException(exc);
         }
     }
 
     @Override
     protected void rangeUpdated() {
-        final double range = getRange();
-        isExponentialForm = range < 1e-3 || range > 1e4;
-        myFormatter.setExponentialForm(isExponentialForm);
-
-        int maxSigDigits = 0;
-        int maxDigits = 0; /* number of digits before separator */
-        if (majorTickMarksCopy != null) {
-            for (int i = 0; i < majorTickMarksCopy.size(); i++) {
-                final double val = majorTickMarksCopy.get(i) / unitScaling;
-                final int nDigits = (int) Math.log10(Math.abs(val)) + 1;
-                maxDigits = Math.max(nDigits, maxDigits);
+        if (majorTickMarksCopy != null && majorTickMarksCopy.size() > 0) {
+            final boolean prevExponentialForm = formatter.isExponentialForm();
+            final int prevPrecision = formatter.getPrecision();
+            configureFormatter(getRange(), majorTickMarksCopy);
+
+            // Clear the cache if the formatting changed
+            if (formatter.isExponentialForm() != prevExponentialForm
+                    || formatter.getPrecision() != prevPrecision
+                    || oldRangeIndex != rangeIndex) {
+                labelCache.clear();
+
+                // TODO: what is the rangeIndex? it does not get set anywhere. Maybe that's a subclass thing?
+                oldRangeIndex = rangeIndex;
```

Comment:
Instead of saving the old state of all properties, we could just increment range index in the setters of the properties. But then again we could also call clear there directly (clearing an already cleared cache shouldn't be expensive).

Reply:
for now I left those lines as they were